src/snowflake/snowpark/dataframe.py

```diff
@@ -1916,15 +1916,12 @@ def join(
         DataFrame and another DataFrame (``right``) on a list of columns
         (``on``).
 
-        The method assumes that the columns in ``how`` have the same meaning
-        in the left and right DataFrames.
-
         Args:
             right: The other :class:`DataFrame` to join.
-            on: A column name or a :class:`Column` object or a list of them to be
-                used for the join. You can also use ``using_columns`` keyword to specify
-                this condition. Note that to avoid breaking changes, currently when
-                ``using_columns`` is specified, it overrides ``on``.
+            on: A column name or a :class:`Column` object or a list of them to be used for the join.
+                When a list of column names are specified, this method assumes the named columns are present in both dataframes.
+                You can use keyword ``using_columns`` to specify this condition. Note that to avoid breaking changes,
```

Comment:
Do we need to change certain line breaks so we don't have very long lines?

Reply:
Personally I did that because I'm too lazy to scroll left and right. But our formatter (black) doesn't check the length of docstring.